paasta_tools/setup_istio_mesh.py

```diff
@@ -192,7 +223,7 @@ def setup_kube_services(
 
     return setup_paasta_namespace_services(
         kube_client, namespaces.keys(), existing_kube_services_names, rate_limit
-    )
+    ) and cleanup_kube_svc(kube_client, namespaces.keys(), existing_kube_services_names)
```

Comment:
rather than just returning `and` can you do something like

```
setup_services = etup_paasta_namespace_services(
        kube_client, namespaces.keys(), existing_kube_services_names, rate_limit
)
delete_services = cleanup_kube_svc(kube_client, namespaces.keys(), existing_kube_services_names)
if not setup_services:
   log.info("failed to setup kubernetes svc objects")
else:
   log.info("succesfully deleted kubernetes svc objects")

if not delete_services:
  ...
else:
  ...
```

that way it'll be clearer to the incident responder what's gone on when it has failed

Reply:
also, maybe you should move the call to `cleanup_kube_svc` to the `main` function, rather than overloading `setup_`

Reply:
The idea behind returning `and` was that , we have already logged the error case in form of exceptions from calls to `setup_paasta_namespace_service` or `cleanup_kube_svc` ( example  https://github.com/Yelp/paasta/blob/340344c3765df36de69468b8bfc4a5d273b2c3bb/paasta_tools/setup_istio_mesh.py#L169 ). At this point we can fail with any non zero error code ( So that we can mark the cron as failure and trigger the desired results).

Reply:
Here we are using `setup_kube_services` as an orchestrator , Since we needed information processed in earlier steps to get re-used ( like list of know namespaces  , getting svc etc). Maybe we need to rename it to something better ( like `process_services`, Still i think it would be preferable to keep the flow of `setup_kube_services`.

Reply:
right, yeah - `setup_kube_services` is a bit of a misnomer. maybe `ensure_kube_services` or something, but it's minor. I'll drop this one